core/src/smartcontracts/isi/asset.rs

```diff
@@ -275,46 +271,49 @@ pub mod isi {
         }
     }
 
+    /// Assert that the two assets have the same asset `definition_id`.
+    fn assert_matching_definitions<W: WorldTrait>(
+        source: &<Asset as Identifiable>::Id,
+        destination: &<Asset as Identifiable>::Id,
+        wsv: &WorldStateView<W>,
+        value_type: AssetValueType,
+    ) -> Result<(), Error> {
+        if destination.definition_id != source.definition_id {
```

Comment:
so the limitation is that it's not possible to transfer assets across domains? and they have to have the same name